### PR TITLE
Noncopyable proposal

### PR DESCRIPTION
**Pull request by nickaleks**
### Description of the Change

Implement non-copyable proposal.

### Benefits

Avoid unnecessary copying of proposal. Separate code to header and cpp results in faster compilation.

### Possible Drawbacks 

None

### CODE REVIEW

File: test/benchmark/bm_proto_creation.cpp [hunk 1]
```diff
@@ -11,8 +11,8 @@
  *
  * The purpose of this benchmark is to keep track of performance costs related
  * to blocks and proposals copying/moving.
- *
- * Each benchmark runs transaction() and commands() call to
+ *
```
Comment (muratovv) [inline test/benchmark/bm_proto_creation.cpp]: Looks like redundant space.

File: test/benchmark/bm_proto_creation.cpp [hunk 2]
```diff
@@ -11,8 +11,8 @@
  *
  * The purpose of this benchmark is to keep track of performance costs related
  * to blocks and proposals copying/moving.
- *
- * Each benchmark runs transaction() and commands() call to
+ * 
+ * Each benchmark runs transaction() and commands() call to
```
Comment (muratovv) [inline test/benchmark/bm_proto_creation.cpp]: sab

File: shared_model/backend/protobuf/proposal.hpp [hunk 3]
```diff
@@ -21,48 +9,38 @@
...
 
      private:
-      // lazy
       template <typename T>
       using Lazy = detail::LazyInitializer<T>;
 
       const Lazy<std::vector<proto::Transaction>> transactions_{[this] {
-        return std::vector<proto::Transaction>(proto_->transactions().begin(),
-                                               proto_->transactions().end());
+        return std::vector<proto::Transaction>(
```
Comment (Akvinikym) [inline shared_model/backend/protobuf/proposal.hpp]: Isn't it possible to move this initialisation to constructor in cpp file as well?

Comment (nickaleks) [reply shared_model/backend/protobuf/proposal.hpp]: Unfortunately no

**Pull request merged**